rate_limiter.rst

```diff
@@ -158,35 +159,59 @@ the number of requests to the API::
     for the :ref:`kernel.request event <component-http-kernel-kernel-request>`
     and check the rate limiter once for all requests.
 
-In other scenarios you may want instead to wait as long as needed until a new
-token is available. In those cases, use the ``wait()`` method::
+Wait until a Token is Available
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Instead of dropping a request or process when the limit has been reached,
+you might want to wait until a new token is available. This can be achieved
+using the ``reserve()`` method::
 
     // src/Controller/ApiController.php
     namespace App\Controller;
 
     use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
     use Symfony\Component\HttpFoundation\Request;
-    use Symfony\Component\RateLimiter\Limiter;
+    use Symfony\Component\RateLimiter\RateLimiter;
 
     class ApiController extends AbstractController
     {
-        public function registerUser(Request $request, Limiter $authenticatedApiLimiter)
+        public function registerUser(Request $request, RateLimiter $authenticatedApiLimiter)
         {
             $apiKey = $request->headers->get('apikey');
             $limiter = $authenticatedApiLimiter->create($apiKey);
 
             // this blocks the application until the given number of tokens can be consumed
-            do {
-                $limit = $limiter->consume(1);
-                $limit->wait();
-            } while (!$limit->isAccepted());
+            $limiter->reserve(1)->wait();
+
+            // optional, pass a maximum wait time (in seconds), a MaxWaitDurationExceededException
+            // is thrown if the process has to wait longer. E.g. to wait at most 20 seconds:
+            //$limiter->reserve(1, 20)->wait();
```

Comment:
Is this example correct? I would expect it to be:

```php
$limiter->reserve(1)->wait(20);
```

Reply:
Unfortunately, yes. I would also prefer your way, but the goal of max time is avoid reserving a token if we know the wait time is longer than its value. So the `reserve()` method must be aware of the max wait time.

Reply:
Thinking out loud: what if we had these methods instead?

```php
reserve(int $tokens = 1);
reserveAndWait(int $tokens = 1, ?float $maxTime = null);


reserve();
reserve(2);
reserveAndWait();
reserveAndWait(2);
reserveAndWait(1, 20);
```

Reply:
In any case, I'm merging your nice PR because this discussion is something to discuss (or not) in the code repository instead of here. Thanks!